platform/ios/src/MLNMapView+Metal.mm

```diff
@@ -0,0 +1,227 @@
+
+#import <mbgl/mtl/renderable_resource.hpp>
+
+#import <MetalKit/MetalKit.h>
+#import <Metal/Metal.h>
+#import <QuartzCore/CAMetalLayer.h>
+
+#import <Metal/Metal.hpp>
+
+@interface MLNMapViewImplDelegate : NSObject <MTKViewDelegate>
+@end
+
+@implementation MLNMapViewImplDelegate {
+    MLNMapViewMetalImpl* _impl;
+}
+
+- (instancetype)initWithImpl:(MLNMapViewMetalImpl*)impl {
+    if (self = [super init]) {
+        _impl = impl;
+    }
+    return self;
+}
+
+- (void)mtkView:(MTKView *)view drawableSizeWillChange:(CGSize)size {}
+
+- (void)drawInMTKView:(MTKView *)view {
+    _impl->render();
+}
+
+@end
+
+namespace {
+CGFloat contentScaleFactor() {
+    return [UIScreen instancesRespondToSelector:@selector(nativeScale)]
+        ? [[UIScreen mainScreen] nativeScale]
+        : [[UIScreen mainScreen] scale];
+}
+} // namespace
+
+class MLNMapViewMetalRenderableResource final : public mbgl::mtl::RenderableResource {
+public:
+    MLNMapViewMetalRenderableResource(MLNMapViewMetalImpl& backend_)
+        : backend(backend_),
+          delegate([[MLNMapViewImplDelegate alloc] initWithImpl:&backend]),
+          atLeastiOS_12_2_0([NSProcessInfo.processInfo
+              isOperatingSystemAtLeastVersion:(NSOperatingSystemVersion){ 12, 2, 0 }]) {
+    }
+
+    void bind() override {
+        if (!commandQueue) {
+            commandQueue = [mtlView.device newCommandQueue];
+        }
+
+        commandBuffer = [commandQueue commandBuffer];
+        commandBufferPtr = NS::RetainPtr((__bridge MTL::CommandBuffer*)commandBuffer);
+
+        currentDrawable = [mtlView currentDrawable];
+    }
+
+    const mbgl::mtl::RendererBackend& getBackend() const override { return backend; }
+
+    const mbgl::mtl::MTLCommandBufferPtr& getCommandBuffer() const override {
+        return commandBufferPtr;
+    }
+
+    virtual mbgl::mtl::MTLBlitPassDescriptorPtr getUploadPassDescriptor() const override {
+        // Create from render pass descriptor?
+        return NS::TransferPtr(MTL::BlitPassDescriptor::alloc()->init());
+    }
+
+    mbgl::mtl::MTLRenderPassDescriptorPtr getRenderPassDescriptor() const override {
+        return NS::RetainPtr((__bridge MTL::RenderPassDescriptor*)mtlView.currentRenderPassDescriptor);
+    }
+
+    void swap() override {
+        [commandBuffer presentDrawable:currentDrawable];
+        [commandBuffer commit];
+        commandBuffer = nil;
+        commandBufferPtr.reset();
+    }
+
+    mbgl::Size framebufferSize() {
+        assert(mtlView);
+        return { static_cast<uint32_t>(mtlView.drawableSize.width),
+                 static_cast<uint32_t>(mtlView.drawableSize.height) };
+    }
+
+private:
+    MLNMapViewMetalImpl& backend;
+    mbgl::mtl::MTLCommandBufferPtr commandBufferPtr;
+
+public:
+    MLNMapViewImplDelegate* delegate = nil;
+    MTKView *mtlView = nil;
+    id<CAMetalDrawable> currentDrawable;
+    id <MTLCommandBuffer> commandBuffer;
+    id <MTLCommandQueue> commandQueue;
+    const bool atLeastiOS_12_2_0;
```

Comment:
Is this even still used somewhere?

Reply:
I don't see any references to it, even in legacy/OpenGL